scripts/release-tools/downstream/y-stream-setup.sh

```diff
@@ -0,0 +1,119 @@
+    #supported_versions=($(curl -fsSL "https://access.redhat.com/product-life-cycles/api/v1/products?name=Red%20Hat%20Advanced%20Cluster%20Security%20for%20Kubernetes" | jq -r '.data[0].versions[] | select(.type == "Full Support") | .name'))
+
+    mapfile -t supported_versions < <(
+      curl -fsSL "https://access.redhat.com/product-life-cycles/api/v1/products?name=Red%20Hat%20Advanced%20Cluster%20Security%20for%20Kubernetes" |
+      jq -r '.data[0].versions[] | select(.type == "Full Support") | .name'
+    )
+
+
+    nversions=${#supported_versions[@]}
+    for ((i = nversions - 1; i >= 0; i = i - 1)); do
+        echo "${supported_versions[$i]}"
+    done
+
+    echo "$release"
+}
+
+update_content_stream_tags() {
+    mapfile -t versions < <(get_supported_versions)
+
+    nversions=${#versions[@]}
+
+    for ((i=0; i<nversions; i++)); do
+        find versions/release-* -name 'product.yml' -exec bash -c 'yq w -i "$1" delivery-repo-content.content_stream_tags[$2] "$3" --style=double' _ {} "$i" "${versions[$i]}" \;
+    done
+}
+
+perform_sed_or_gsed() {
+    local search="$1"
+    local replace="$2"
+    local file="$3"
+
+    if command -v sed > /dev/null; then
+        sed -i "s|$search|$replace|" "$file"
+    elif command -v gsed > /dev/null; then
+        gsed -i "s|$search|$replace|" "$file"
+    else
+        echo "Error: Neither sed nor gsed found. Cannot perform replacement."
+        exit 1
+    fi
+}
+
+export -f perform_sed_or_gsed
+
+git clone git@gitlab.cee.redhat.com:cpaas-products/rhacs.git gitlab-rhacs
+pushd gitlab-rhacs
+
+git checkout -b "setup-${release}"
+
+cp -R "versions/release-${previous_release}" "versions/release-${release}"
+
+pushd "versions/release-${release}"
+
+rm -f advisory_map.yml
+
+# Update release.yml with the correct version. Sinple sed should be safe, but check changes
+sed -i "s|$previous_release|$release|" release.yml
+
+# Update product.yml with the correct version.
+# This is a little more complicated since not all occurances of the old release should be changed
+# Check changes
+perform_sed_or_gsed "rhacs-$previous_release" "rhacs-$release" product.yml
+perform_sed_or_gsed "RHACS-$previous_release" "RHACS-$release" product.yml
+perform_sed_or_gsed "RHACS $previous_release" "RHACS $release" product.yml
+perform_sed_or_gsed "Kubernetes $previous_release" "Kubernetes $release" product.yml
+
+yq w -i product.yml product.release.version "${release}.0" --style=single
+yq w -i product.yml honeybadger.version "${release}" --style=single
+yq w -i product.yml product.honeybadger.version "${release}" --style=single
+
+popd
+
+# Add the versions to content_stream_tags in all product.yml files
+update_content_stream_tags
+
+# yq makes some unwanted changes that need to be undone
+find versions/release-* -name 'product.yml' -exec bash -c 'perform_sed_or_gsed "!!merge " "" "$1"' _ {} \;
+#find versions/release-* -name 'product.yml' -exec bash -c "perform_sed_or_gsed '!!merge ' '' '{}'" \;
```

Comment:
This looks a bit sloppy with undoing the changes and using a `*` in find. I think this will be a problem in the longer term.

Shall we maybe create a template instead where the versions are placeholders? We could then instantiate the template with a proper version instead of dong search & replace.

Reply:
I am not sure if this is a good idea or not. Both methods have pros and cons. With the templates I would be concerned that someone will make a change to a file and then not change the template, and then when the template is used the change is lost. With the method in this PR I am concerned that this script is brittle.